### PR TITLE
FX2 speed test

### PR DESCRIPTION
**Pull request by martinling**
Stacked on #4.

This PR adds a USB HS speed test using the Cypress FX2 on Tycho connected via the Cynthion EUT's TARGET passthrough.

The FX2 firmware used is an adaptation of the "Cystream" code from the vendor, built from [this repository](https://github.com/shenki/cystream). This version includes a number of optimizations which are discussed in [this blog post](https://allmybrain.com/2009/04/14/fx2-cystream-throughput-test-with-sdcc-and-fx2lib/).

I do not see the throughput reported in that blog post on my test setup. A typical result for me is 37.7MB/s. Perhaps this can be improved somehow, but it's sufficient to prove that the passthrough is operating at HS speeds. I have set the expected speed range to between 35 and 45 MB/s.

Test output:
```
 === Please connect cable to EUT Target-A port and press PASS or FAIL === 

• Checking Target-A cable is connected:
  • Setting DC-DC converter to 5.00 V 0.25 A
  • Connecting DC-DC converter to TARGET-C
  • Checking TARGET-A cable is connected:
    • Checking voltage on VBUS_TA is within 4.85 to 5.05 V: 5.00 V
• Testing Target-C to Target-A data passthrough:
  • Connecting host D+/D- to TARGET-C... OK
  • Checking for FX2:
    • Looking for device with:
      • VID: 0x04b4, PID: 0x8613
      • Manufacturer: None
      • Product: None
      • Found at bus 2 address 6:
        • Device serial is None
  • Loading FX2 firmware... OK
  • Looking for device with:
    • VID: 0x04b4, PID: 0x1003
    • Manufacturer: None
    • Product: Cy-stream
    • Found at bus 2 address 7:
      • Checking product is Cy-stream... OK
      • Device serial is None
  • Running speed test... OK
  • Checking transfer rate on TARGET-C is within 35.00 to 45.00 MB/s: 37.70 MB/s
  • Disconnecting DC-DC converter
```

### CODE REVIEW

**Comment by grvvy** [comment]
My setup fails the added `TARGET_PHY_1V8` check.
```
• Checking all PHY supply voltages:
  • Checking voltage on CONTROL_PHY_3V3 is within 3.25 to 3.35 V: 3.31 V
  • Checking voltage on CONTROL_PHY_1V8 is within 1.75 to 1.85 V: 1.82 V
  • Checking voltage on AUX_PHY_3V3 is within 3.25 to 3.35 V: 3.31 V
  • Checking voltage on AUX_PHY_1V8 is within 1.75 to 1.85 V: 1.78 V
  • Checking voltage on TARGET_PHY_3V3 is within 3.25 to 3.35 V: 3.31 V
  • Checking voltage on TARGET_PHY_1V8 is within 1.75 to 1.85 V: 1.72 V

FAIL: voltage too low on TARGET_PHY_1V8: 1.719 V, minimum was 1.75 V
```

File: eut.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
     ('AUX_PHY_3V3',     3.25, 3.35),
     ('AUX_PHY_1V8',     1.75, 1.85),
     ('TARGET_PHY_3V3',  3.25, 3.35),
-    ('TARGET_PHY_3V3',  3.25, 3.35))
+    ('TARGET_PHY_1V8',  1.75, 1.85))
```
Comment (grvvy) [inline eut.py]: I had to lower the minimum to get this working on my setup
` ('TARGET_PHY_1V8',  1.70, 1.85))`

File: Makefile [hunk 2]
```diff
@@ -24,6 +24,15 @@ debug: $(TIMESTAMP)
 calibrate: $(TIMESTAMP)
 	$(ENV_PYTHON) calibrate.py
 
+check: $(TIMESTAMP)
+	$(ENV_PYTHON) check.py
+
+install-udev: 60-tycho.rules
+	sudo cp $< /etc/udev/rules.d/
+
```
Comment (grvvy) [inline Makefile]: Worth adding a reload after copying the udev rules for convenience?
`sudo udevadm control --reload-rules && sudo udevadm trigger`

**Review by grvvy — APPROVED** [review]

File: eut.py (hunk 1 above)
Comment (martinling) [reply eut.py]: Let's adjust those limits in a separate PR.

File: Makefile (hunk 2 above)
Comment (martinling) [reply Makefile]: I haven't found this necessary; as I understand it `udevd` is supposed to monitor the rules directory automatically with inotify.

**Pull request merged**